File: BackEnd/PHP/laravelshop_eloquentorm/tests/E2E/OpsDashboardDependencyFailuresE2ETest.php
<?php

namespace Tests\E2E;

use Illuminate\Foundation\Testing\RefreshDatabase;
use Tests\Support\ApiAuthTestHelpers;
use Tests\TestCase;

class OpsDashboardDependencyFailuresE2ETest extends TestCase
{
    use ApiAuthTestHelpers;
    use RefreshDatabase;

    public function test_ops_dashboard_exposes_degraded_state_when_dependency_checks_are_observed(): void
    {
        $admin = $this->createApiUser('ADMIN');

        $this->actingAsApiUser($admin)
            ->getJson('/api/v1/admin/ops-dashboard/summary')
            ->assertOk()
            ->assertJsonPath('data.overallStatus', 'degraded');

        $this->actingAsApiUser($admin)
            ->getJson('/api/v1/admin/observability/dashboard')
            ->assertOk()
            ->assertJsonPath('data.opsSummary.overallStatus', 'degraded');
    }
}

File: BackEnd/Java/java-spring-maven-jpa-postgresql/src/test/java/com/pbshop/java/spring/maven/jpa/postgresql/e2e/OpsDashboardDependencyFailuresE2ETest.java
package com.pbshop.java.spring.maven.jpa.postgresql.e2e;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import com.pbshop.java.spring.maven.jpa.postgresql.searchsync.SearchIndexOutbox;
import com.pbshop.java.spring.maven.jpa.postgresql.searchsync.SearchIndexOutboxRepository;
import com.pbshop.java.spring.maven.jpa.postgresql.support.AuthenticatedApiIntegrationSupport;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class OpsDashboardDependencyFailuresE2ETest extends AuthenticatedApiIntegrationSupport {

    @Autowired
    private SearchIndexOutboxRepository searchIndexOutboxRepository;

    @Test
    void opsDashboardShowsDependencyFailureSignals() throws Exception {
        String adminToken = createUserAndLogin("ADMIN");

        SearchIndexOutbox outbox = new SearchIndexOutbox();
        outbox.setEntityType("PRODUCT");
        outbox.setEntityId(1L);
        outbox.setStatus("FAILED");
        searchIndexOutboxRepository.save(outbox);

        mockMvc.perform(get("/api/v1/admin/ops-dashboard/summary")
                        .header("Authorization", "Bearer " + adminToken))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.data.overallStatus").value("degraded"))
                .andExpect(jsonPath("$.data.queue.items[2].counts.failed").value(1));
    }
}
